src/Illuminate/Database/Eloquent/Relations/Concerns/ComparesRelatedModels.php

```diff
@@ -65,4 +67,25 @@ protected function compareKeys($parentKey, $relatedKey)
 
         return $parentKey === $relatedKey;
     }
+
+    /**
+     * Determine if the given model is the correct relationship model.
```

Comment:
What is "correct" here? I don't feel like the method name or its description is very informative.

Reply:
True, the description and name are indeed not very accurate/appropriate. The method is also not needed, since the implementation remains the same for other relations implementing `PartialRelation`. The following should be more clear:

```php
    public function is($model)
    {
        $match = ! is_null($model) &&
               $this->compareKeys($this->getParentKey(), $this->getRelatedKeyFrom($model)) &&
               $this->related->getTable() === $model->getTable() &&
               $this->related->getConnectionName() === $model->getConnectionName();
        
        if(! $match) {
            return false;
        }
        
        if(! $this instanceof PartialRelation && ! $this->isOneOfMany()) {
            return $match;
        }
        
        // For "one-of-many" relationships, existence must be checked since keys 
        // also match for models that are not the related instance of the relationship.
        return $this->query
            ->whereKey($model->getKey())
            ->exists();
    }
```

Reply:
So, if the method is not needed - what should I do? :)

Reply:
I will remove the method and implement the code shown above. I did not want to implement it right away in case there are objections 😅